### PR TITLE
docs: improve how to use `<details>`

### PR DESCRIPTION
**Pull request by tats-u**
## Pre-flight checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [ ] **If this is a code change**: I have written unit tests and/or added dogfooding pages to fully verify the new behavior.
- [ ] **If this is a new API or substantial change**: the PR has an accompanying issue (closes #0000) and the maintainers have approved on my working plan.

## Motivation

The current example for `<details>` is not ideal one.

- There are extra `<div>` elements, which confuse readers (You don't have to use them; they. They're just optional and not mandatory)
- The contents in `<div>` confuse the block and inline notations

Block notation:

```mdx
<div>
  paragraph inside `<div>` element
</div>
```

Inline notation:

```mdx
<div>no paragraphs in this line</div>
```

## Test Plan

https://deploy-preview-10173--docusaurus-2.netlify.app/docs/markdown-features#details
https://deploy-preview-10173--docusaurus-2.netlify.app/docs/3.3.2/markdown-features#details

### Test links

Deploy preview: https://deploy-preview-10173--docusaurus-2.netlify.app/

### CODE REVIEW

File: website/docs/guides/markdown-features/markdown-features-intro.mdx [hunk 1]
```diff
@@ -181,42 +181,105 @@ Markdown quotes are beautifully styled:
...
+    <summary>Nested toggle! Some surprise inside...</summary>
+
+    😲😲😲😲😲
+  </details>
 </details>
 
 </BrowserWindow>
+````
+
+:::warning
```
Comment (slorber) [inline website/docs/guides/markdown-features/markdown-features-intro.mdx:232]: this section is way too big and visible

**Review by slorber — CHANGES_REQUESTED** [review]
I'd be ok to modify the example

However we try to keep the docs short and concise, not to explain every MDX quirk in depth. There's the MDX docs for that. 

The warning is way too visible and draw way too much attention to it while most readers shouldn't care much. I'd prefer to remove the warning entirely. If you want to keep it, make it short (max 3 lines) and add a link to relevant MDX docs page. If it's not documented on MDX, make a doc PR to MDX instead. If details must be included, hide them by default in a `<details>` element.

You see the idea. The goal is not for the docs to become too long because every one document every quirk ever encountered 😅

**Comment by OzakIOne** [comment]
image

image

Syntax color needs to be fixed

**Comment by tats-u** [comment]
Looks like Docusaurus (Prism itself?) doesn't support the highlight for `mdx`.

**Comment by slorber** [comment]
Thanks, still think it's a bit too much but let's move on

**Comment by tats-u** [comment]
Oh, I've been planning to fix it if you suggest me a changing plan. I think it probably can be a little more shortened with others' help.
Sorry for falling asleep before leaving comments before the merge.
Of course I'm glad to know this PR is merged and I'll tell you thank you for merging, though...

**Comment by tats-u** [comment]
I think I have to send a PR to MDX's document as you pointed out.
This topic is written only in [the migration guide to v2](https://mdxjs.com/migrating/v2/), but it's difficult for us to arrive from the top; We have to go to Blog → MDX 2 post before it.

**Comment by tats-u** [comment]
I shortened notes
I think the official MDX isn't friendly for intermediates, so I'll have to send a PR there....

Also, I fell asleep before pushing the backport change to 3.3 to GitHub...

**Pull request merged**